src/core/balance-sheet.ts

```diff
@@ -1,29 +1,55 @@
 import { getBalance as getBitcoindBalance } from "@services/bitcoind"
 import { lndsBalances } from "@services/lnd/utils"
 import { baseLogger } from "@services/logger"
-import { User } from "@services/mongoose/schema"
+import { InvoiceUser, User } from "@services/mongoose/schema"
 import { ledger } from "@services/mongodb"
 
 import { WalletFactory } from "./wallet-factory"
 
 const logger = baseLogger.child({ module: "balanceSheet" })
 
+export const updatePendingLightningTransactions = async () => {
+  let user, userWallet
+
+  // select distinct user ids from pending invoices
+  const usersWithPendingInvoices = InvoiceUser.aggregate([
+    { $match: { paid: false } },
+    { $group: { _id: "$uid" } },
+  ])
+    .cursor({ batchSize: 100 })
+    .exec()
+
+  for await (const { _id } of usersWithPendingInvoices) {
```

Comment:
does this loop one by one? or 100 by 100 in parrallel? 
I guess it's one by one, but only have 100 InvoiceUser in memory at one particular time?

Reply:
we could speed this up by making those call in parrallel maybe with batch of 5. most of the underlying logic here is waiting for network calls on external service.

Reply:
this one `I guess it's one by one, but only have 100 InvoiceUser in memory at one particular time`

Reply:
maybe let's do that in another PR

Reply:
Fixed https://github.com/GaloyMoney/galoy/pull/421/commits/1b4e48a0a66955743533e8717949f8df580aee5d